src/cli/examples/llvm-source-coverage/README.md

```diff
@@ -0,0 +1,134 @@
+# Performing Source-Coverage using Analysis Tasks
+
+The `generic_analysis` task can be used to perform user-defined analysis for every input.  
+
+Running an application compiled with [LLVM's source-based code coverage](https://clang.llvm.org/docs/SourceBasedCodeCoverage.html) with each input can be used to generate source based coverage information.
```

Comment:
```suggestion
Running an application compiled with [LLVM's source-based code coverage](https://clang.llvm.org/docs/SourceBasedCodeCoverage.html) with each input can be used to generate source based coverage information. In this case, the core of the analysis is performed via compiled-in target instrumentation, but this is not required.
```

Reply:
Can you explain your intent for this addition?  The compiled-in instrumentation is required for this example.

Reply:
This is just a suggestion for emphasizing the big picture of "analysis tasks" while explaining this specific task (and is not essential).

The "analyzer" in this example is just a bit odd for the following reason: it doesn't perform analysis per se. It is just a script that mostly just invokes the `target_exe` and juggles files that the `target_exe` itself generates. Per its name, and in typical scenarios, I think we expect that the `analyzer_exe` is usually a program that _creates_ the analysis files. This larger example is just a bit unusual in the way that libFuzzers are unusual: the actual _analyzer_ is baked into the target here, just like a libFuzzer has the fuzzer baked into the target as instrumentation.

The suggested comment about that might be more confusing than clarifying in context, though.

Reply:
In this case, the analyzer_exe does create the files by calling first the fuzzer, then llvm-profdata, then llvm-cov.

Reply:
So, to be clear, there's no edit needed here, and I think this is one of the best docs we (will) have. The above was just my original rationale about emphasizing what's special/slightly unusual in this example.